### PR TITLE
Enhancing ITP to accept email-based accounts.

### PR DESCRIPTION
**Pull request by alxdarksage**

### CODE REVIEW

File: src/test/java/org/sagebionetworks/bridge/sdk/integration/IntentToParticipateTest.java [hunk 1]
```diff
@@ -120,11 +121,66 @@ public void testIntentToParticipate() throws Exception {
             
             ConsentStatus status = session.getConsentStatuses().get(IntegTestUtils.STUDY_ID);
             assertTrue(status.isConsented());
+        } catch(Exception e) {
+            fail("This should not have thrown an exception");
```
Comment (DwayneJengSage) [inline src/test/java/org/sagebionetworks/bridge/sdk/integration/IntentToParticipateTest.java]: Don't do this. This masks any exception thrown by the test as well as any assert.

File: src/test/java/org/sagebionetworks/bridge/sdk/integration/IntentToParticipateTest.java [hunk 2]
```diff
@@ -120,11 +121,66 @@ public void testIntentToParticipate() throws Exception {
...
+            
+            // This does not throw a consent exception.
+            AuthenticationApi authApi = provider.getClient(AuthenticationApi.class);
+            UserSessionInfo session = authApi.signInV4(user.getSignIn()).execute().body();
+            assertEquals(SharingScope.ALL_QUALIFIED_RESEARCHERS, session.getSharingScope());
+            
+            ConsentStatus status = session.getConsentStatuses().get(IntegTestUtils.STUDY_ID);
+            assertTrue(status.isConsented());
+        } catch(Exception e) {
+            fail("This should not have thrown an exception");
```
Comment (DwayneJengSage) [inline src/test/java/org/sagebionetworks/bridge/sdk/integration/IntentToParticipateTest.java]: Similarly here.

**Review by DwayneJengSage — CHANGES_REQUESTED** [review]

**Review by DwayneJengSage — CHANGES_REQUESTED** [review]
Also, do we want to add appInstallLinkTemplate to StudyTest.crudStudy()?

File: src/test/java/org/sagebionetworks/bridge/sdk/integration/IntentToParticipateTest.java (hunk 1 above)
Comment (alxdarksage) [reply src/test/java/org/sagebionetworks/bridge/sdk/integration/IntentToParticipateTest.java]: Removed

File: src/test/java/org/sagebionetworks/bridge/sdk/integration/IntentToParticipateTest.java (hunk 2 above)
Comment (alxdarksage) [reply src/test/java/org/sagebionetworks/bridge/sdk/integration/IntentToParticipateTest.java]: Removed

**Review by DwayneJengSage — APPROVED** [review]

**Pull request merged**